tests/pipelines/self_segmentation_stable_diffusion/test_self_segementation_stable_diffusion.py

```diff
@@ -0,0 +1,261 @@
+        device = "cpu"  # ensure determinism for the device-dependent torch.Generator
+        components = self.get_dummy_components()
+        sd_pipe = SelfSegmentationStableDiffusionPipeline(**components)
+        sd_pipe = sd_pipe.to(device)
+        sd_pipe.set_progress_bar_config(disable=None)
+
+        inputs = self.get_dummy_inputs(device)
+        pipe_out = sd_pipe(**inputs)
+        image = pipe_out.images
+        seg_map = pipe_out.seg_map[0]
+        seg_labels = pipe_out.seg_labels[0]
+        merged_seg_map = pipe_out.merged_seg_map[0]
+        merged_seg_labels = pipe_out.merged_seg_labels[0]
+
+        assert image.shape == (1, 128, 128, 3)
+        assert seg_map.shape == merged_seg_map.shape == (32, 32)
+        assert len(seg_labels) == 6
+        assert len(merged_seg_labels) <= len(seg_labels)
+
+        expected_slice = np.array([0.4672, 0.4133, 0.4557, 0.4856, 0.4410, 0.5083, 0.4741, 0.4604, 0.4592])
+        image_slice = image[0, -3:, -3:, -1]
+        assert np.abs(image_slice.flatten() - expected_slice).max() < 1e-2
+
+    def test_self_segmentation_stable_diffusion_negative_prompt(self):
+        device = "cpu"  # ensure determinism for the device-dependent torch.Generator
+        components = self.get_dummy_components()
+        sd_pipe = SelfSegmentationStableDiffusionPipeline(**components)
+        sd_pipe = sd_pipe.to(device)
+        sd_pipe.set_progress_bar_config(disable=None)
+
+        inputs = self.get_dummy_inputs(device)
+        negative_prompt = "french fries"
+        output = sd_pipe(**inputs, negative_prompt=negative_prompt)
+        image = output.images
+        image_slice = image[0, -3:, -3:, -1]
+
+        assert image.shape == (1, 128, 128, 3)
+
+        expected_slice = np.array([0.4619, 0.4208, 0.4553, 0.4734, 0.4368, 0.5026, 0.4690, 0.4550, 0.4567])
+
+        assert np.abs(image_slice.flatten() - expected_slice).max() < 1e-2
+
+    def test_self_segmentation_stable_diffusion_batch(self):
+        device = "cpu"  # ensure determinism for the device-dependent torch.Generator
+        components = self.get_dummy_components()
+        sd_pipe = SelfSegmentationStableDiffusionPipeline(**components)
+        sd_pipe = sd_pipe.to(device)
+        sd_pipe.set_progress_bar_config(disable=None)
+
+        inputs = self.get_dummy_inputs(device)
+        inputs["prompt"] = [inputs["prompt"]] * 3
+        output = sd_pipe(**inputs)
+        image = output.images
+        seg_maps = output.seg_map
+        seg_labels = output.seg_labels
+        merged_seg_maps = output.merged_seg_map
+        merged_seg_labels = output.merged_seg_labels
+
+        assert image.shape == (3, 128, 128, 3)
+        assert len(seg_maps) == len(seg_labels) == 3
+        assert len(merged_seg_maps) == len(merged_seg_labels) == 3
+
+    def test_float16_inference(self, expected_max_diff=1e-2):
+        pass
+
+    def test_karras_schedulers_shape(self):
+        pass
+
```

Comment:
Do we know why this does not work?

Reply:
My pipeline relies on an attention processor that saves attention maps. After all the denoising steps, I use these maps to segment. For some reason, in these two tests the attention maps that I save disappear after the unet returns its output (at each denoising step). For the float16, I tested the pipeline and it does work.

Reply:
I think it's fine then to disable them for now.